modules/util/portal-tools-wsdd-builder/src/main/java/com/liferay/portal/tools/wsdd/builder/WSDDBuilder.java

```diff
@@ -83,7 +83,7 @@ public void build() throws Exception {
 		SAXReader saxReader = _getSAXReader();
 
 		Document document = saxReader.read(
-			new XMLSafeReader(ToolsUtil.getContent(_fileName)));
+			new UnsyncStringReader(ToolsUtil.getContent(_fileName)));
```

Comment:
@julschong Why you can do this? Please figure out a reason.

Reply:
@tinatian Hi Tina,
The benefit of using XMLSafeReader is that any character besides '<' passed in before the prolog tag is removed before processing. The reason that this change works here is because the xmls I tested did not have any preceeding characters.
Example:
> "These character will be removed"<?xml version="1.0"?>
> 
> <suppressions>
> 	<source-check>
> 		<suppress checks="JavaXMLSecurityCheck" files="petra-xml/src/main/java/com/liferay/petra/xml/Dom4jUtil\.java" />
> 	</source-check>
> </suppressions>